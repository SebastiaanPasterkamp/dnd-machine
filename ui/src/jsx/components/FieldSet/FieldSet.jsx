import React from 'react';
import PropTypes from 'prop-types';

export const FieldSet = function({ label, children })
{
    return (
        <fieldset className="nice-form-group">
            <legend>{ label }</legend>
            { children }
        </fieldset>
    );
};

FieldSet.propTypes = {
    label: PropTypes.string.isRequired,
    children: PropTypes.any,
};

FieldSet.defaultProps = {
    label: '',
}

export default FieldSet;
