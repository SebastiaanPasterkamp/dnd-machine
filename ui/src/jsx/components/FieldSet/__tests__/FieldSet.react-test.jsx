import React from 'react';
import renderer from 'react-test-renderer';

import FieldSet from '..';

describe('Component: FieldSet', () => {
    it('should render', () => {
        const tree = renderer.create(
            <FieldSet label="test">
                Foo bar
            </FieldSet>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
