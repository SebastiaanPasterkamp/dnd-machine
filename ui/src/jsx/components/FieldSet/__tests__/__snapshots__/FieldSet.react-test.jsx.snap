// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FieldSet should render 1`] = `
<fieldset
  className="nice-form-group"
>
  <legend>
    test
  </legend>
  Foo bar
</fieldset>
`;
